benchmarks/generate_benchmark_params.py

```diff
@@ -0,0 +1,521 @@
+"""Benchmarking parameters generation file."""
+
+import argparse
+from itertools import chain, product
+
+import pandas as pd
+
+parser = argparse.ArgumentParser(
```

Comment:
What are the pros/cons of using argument parser as opposed to a config file?

I personally find config.yml file more intuitive and readable, but I wonder what the community recommends there.

Reply:
Either could be fine. for example pytorch uses command line args. Happy to change it if you prefer otherwise. Thanks!